### PR TITLE
2.16 release notes

### PR DESCRIPTION
**Pull request by LDrago27**
### Description

The PR adds 2.16 Release Notes and changelogs. 

## Changelog

- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/dev/generate_release_note.ts [hunk 1]
```diff
@@ -73,8 +73,17 @@ async function readFragments() {
 
     validateFragment(fragmentContents);
 
-    const fragmentYaml = loadYaml(fragmentContents) as Changelog;
+    const fragmentContentLines = fragmentContents.split('\n');
```
Comment (ananzh) [inline src/dev/generate_release_note.ts:76]: thx for the fix

File: release-notes/opensearch-dashboards.release-notes-3.0.0.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+# VERSION 3.0.0 Release Note
```
Comment (ananzh) [inline release-notes/opensearch-dashboards.release-notes-3.0.0.md:1]: should this be 2.16.0?

File: release-notes/opensearch-dashboards.release-notes-3.0.0.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+### 💥 Breaking Changes
+
+### Deprecations
+
+ - Remove data enhancements config and readonly flag. Removes dead url link, ([#7291](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7291))
+
+### 🛡 Security
+
+ - [CVE-2024-28863] Bump tar from 6.1.11 to 6.2.1 ([#6492](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6492))
```
Comment (ananzh) [inline release-notes/opensearch-dashboards.release-notes-3.0.0.md:11]: nit: I see there are styles inconsistent 
[CVE-2024-28863] Bump tar from `6.1.11` to `6.2.1`

Comment (ananzh) [reply release-notes/opensearch-dashboards.release-notes-3.0.0.md:11]: maybe we could define a style like the above and ask LLM to fix the rest and check typos.

**Review by ananzh — APPROVED** [review]

**Comment by ananzh** [comment]
Do we need to include other PRs that are still not merged in? Let's add them tmrw?

**Comment by ruanyl** [comment]
Should release notes generated on release branch(2.16) and merged to release branch first and then backport to `main`? Are you generating the release notes on `main` branch?

**Comment by LDrago27** [comment]
Closing it will open a new PR with updated Notes

**Pull request closed without merging**